episodes/filetypes-and-metadata.md

```diff
@@ -0,0 +1,498 @@
+opened in many pieces of open source software. Napari-aicsimageio (that we used 
+earlier in this episode) integrates with Bio-Formats to allow many different 
+file formats to be opened in Napari. Bio-Formats is really essential to allow us 
+to work with these multitude of formats! Even so, it won't support absolutely 
+everything, so you will likely need to convert your data to another file format 
+sometimes. If so, it's good practice to use an open file format whose 
+specification is freely available, and can be opened in many different pieces of 
+software e.g. OME-TIFF. 
+
+### Compression
+
+Different file formats use different types of 'compression'. Compression is a 
+way to reduce image file sizes, by changing the way that the image pixel values 
+are stored. There are two main types:
+
+- **Lossless compression** - when the image file is written, an algorithm is 
+applied to reduce its file size. This process is 'lossless' i.e. when the file 
+is opened again the compression process can be reversed perfectly to give 
+exactly the same pixel values.
+
+- **Lossy compression** - when the image is written, an algorithm is applied to 
+reduce its file size. This process is 'lossy' i.e. when the file is opened again 
+the pixel values will be _different_. Some image quality is lost via this 
+process, but it can achieve much smaller file sizes.
+
+For microscopy data, you should therefore use a file format with no compression, 
+or lossless compression. Lossy compression should be avoided as it degrades the 
+pixel values!
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Compression
+
+Let's remove all layers from the Napari viewer, and open the 
+'mitocheck.ome.tiff' dataset. You should have already downloaded this as part of 
+the [setup instructions](../learners/setup.md).
+
+Run the code below in Napari's console. This will save a specific timepoint of 
+this image (time = 30) as four different file formats (.tiff, .png + low and 
+high quality .jpg). Note: these files will be written to the same folder as the 
+'mitocheck.ome.tiff' image!
+
+```python
+from skimage.io import imsave
+from pathlib import Path
+
+# Get the mitocheck.ome layer, and get timepoint = 30
+layer = viewer.layers["mitocheck.ome"]
+image = layer.data[30, :, :]
+
+# Save as different file formats in same folder as mitocheck.ome
+folder_path = Path(layer.source.path).parent
+imsave( folder_path / "test-tiff.tiff", image)
+imsave( folder_path / "test-png.png", image)
+imsave( folder_path / "test-jpg-high-quality.jpg", image, quality=75)
+imsave( folder_path / "test-jpg-low-quality.jpg", image, quality=30)
+
+```
+
+- Go to the folder where 'mitocheck.ome.tiff' was saved and look at the file 
+sizes of the newly written images (`test-tiff`, `test-png`, 
+`test-jpg-high-quality` and `test-jpg-low-quality`). Which is biggest? 
+Which is smallest?
+
+- Open all four images in Napari. Zoom in very close to a bright nucleus, and 
```

Comment:
I had a quick go at creating a difference image of 2 layers in napari but failed. That might be too advanced for this lesson but would be a good way of highlighting what's lost in compression.

Reply:
This worked for me, but the intensity values are quite low (most less than 50) in that slice, so I don't think it's really that helpful, since the differences are only +/- 1 or 2 in each voxel:

```python
img_tiff=viewer.layers['test-tiff']
img_jpg_lo=viewer.layers['test-jpg-low-quality']
#Since both unsigned ints, need to do a cast to avoid clipping/wrapping
diff_jpg_lo=img_tiff.data.astype(int) - img_jpg_lo.data.astype(int)
viewer.add_image(diff_jpg_lo)
```